scripts/provision.sh

```diff
@@ -222,6 +238,18 @@ generate_bldr_keys() {
   hab file upload "builder-api.default" "$(date +%s)" "/hab/cache/keys/${KEY_NAME}.box.key"
 }
 
+is_minio_migration_needed() {
+  installed_version=$(hab pkg path core/minio | cut -d '/' -f6)
+
+  if [ "$installed_version" = "$LATEST_MINIO_VERSION" ]; then
+    return 0
+  elif [[ "$installed_version" < "$LATEST_MINIO_VERSION" ]]; then
+    return 1
+  else
+    return 0
+  fi
```

Comment:
What if the customer changes their `BLDR_CHANNEL` to `on-prem-stable`? In that case they are staying on a non breaking version. We need to check what version of `habitat/builder-minio`  is in `BLDR_CHANNEL`.